### ISSUE TITLE
Do the state renames that were not done in #6762

### ISSUE THREAD

**Issue by oxarbitrage** [ZcashFoundation/zebra#6793]
## Motivation

In #6762 we renamed a lot of stuff from the tasklist in https://github.com/ZcashFoundation/zebra/issues/6680

However there are more things we want to rename, for example some methods, but we don't want #6762 to keep growing. So here is a new task list to do after #6762 gets merged.

The verification parts of the request, state service, and consensus rule checks use incorrect terms that are very confusing for developers and auditors. The storage parts use those terms correctly and should be left alone.

These renames can be implemented and reviewed easily using a `sed` script:
https://github.com/ZcashFoundation/zebra/blob/main/book/src/dev/mass-renames.md#universal-renames-with-sed

### Possible Renames

Checkpoint Blocks in zebra-consensus:
- [x] Rename `VerifyCheckpointError::CommitFinalized` variant to `CommitCheckpointVerified`
- [x] Rename `commit_finalized_block` variable to `commit_checkpoint_verified`

Finalized Queues:
- [x] Rename `drain_queue_and_commit_finalized` method to `drain_finalized_queue_and_commit`
- [x] Rename `queue_and_commit_finalized` method to `queue_and_commit_to_finalized_state`
- [x] Rename `queued_finalized_blocks` to `finalized_state_queued_blocks`
- [x] Replace "queued finalized block" with "finalized state queue block" in comments and logs
- [x] Rename `max_queued_finalized_height` to `max_finalized_queue_height`
- [x] Rename `send_finalized_block_error` method to `send_checkpoint_verified_block_error`
- [x] Rename `QueuedFinalized` to `QueuedCheckpointVerified`

Non-Finalized Queues:
- [x] Rename `queue_and_commit_non_finalized` to `queue_and_commit_to_non_finalized_state`
- [x] Rename `queued_non_finalized_blocks` to `non_finalized_state_queued_blocks`
- [x] Replace "queued non-finalized block" with "non-finalized state queue block" in comments and logs
- [x] Rename `send_non_finalized_block_error` to `send_semantically_verified_block_error`
- [x] Rename `QueuedNonFinalized` to `QueuedSemanticallyVerified`

Channels:
- [x] Rename `last_sent_finalized_block_hash` to `finalized_block_write_last_sent_hash`
- [x] Rename `sent_non_finalized_block_hashes` to `non_finalized_block_write_sent_hashes`
- [x] Rename `invalid_block_reset_receiver` to `invalid_block_write_reset_receiver`

It's also ok to use `semantically_verified` instead of `non_finalized_state`, and `checkpoint_verified` instead of `finalized_state`. But if we do that we also need to change the word order and grammar.

**Comment by teor2345** [comment]
> * Sort out what to do here and update this description. https://github.com/ZcashFoundation/zebra/pull/6762#discussion_r1207067920

I split this out into ticket #6912, because it requires code changes (a struct needs to turn into an enum).

**Comment by teor2345** [comment]
I looked through the verification parts of Zebra, and made a list of all the replacements we could do. We can pick different names, but they should all be consistent with each other.

These renames can be done automatically using `sed`, and reviewed the same way.

@arya2 can you do a quick check of the list?
Feel free to change what we should rename, or what the new names should be.
